### PR TITLE
settings: Error if setting xref doesn't exist

### PR DESCRIPTION
**Pull request by slusarz**
Non-existent reference, e.g. ":dovecot_core:ref:`foobar`", now gives:

Warning, treated as error:
/home/slusarz/dovecot/documentation/source/admin_manual/logging.rst:9:Missing cross-reference: 'foobar'
make: *** [Makefile:19: html] Error 2

### CODE REVIEW

**Review by cmouse — APPROVED** [review]

**Review by cmouse — CHANGES_REQUESTED** [review]
should the comments be fixed?

File: source/_ext/dovecot_sphinx.py [hunk 1]
```diff
@@ -177,6 +178,8 @@ def resolve_xref(self, env, fromdocname, builder, typ, target, node,
       return make_refnode(builder, fromdocname, todocname, targ, contnode,
                           targ)
     else:
+      logger = logging.getLogger(__name__)
+      logger.warning('Missing cross-reference: %r', target, location=node)
```
Comment (mrannanj) [inline source/_ext/dovecot_sphinx.py:182]: Is there a `logger.error()` that this could use?

Comment (slusarz) [reply source/_ext/dovecot_sphinx.py:182]: logger.warning() is the expected behavior, as Sphinx will emit an event to allow the code to potentially fix/address this.  For our purposes, we want warnings to be treated as errors - but that's run-time configuration.

**Comment by slusarz** [comment]
All pre-existing errors that were exposed by this change (causing warnings) have now been fixed.

**Review by cmouse — APPROVED** [review]

**Pull request merged**